Creation of Landtrendr.AGB.Model [Work In Progress]

```diff
@@ -0,0 +1,203 @@
+#' @export
+#' 
+#' @examples
+#' @author Tami Gordon [02/08/24]
+#' @importFrom magrittr %>%
+#' 
+---
+
+```{r setup, include= TRUE}
+knitr::opts_chunk$set(echo = TRUE)
+## Getting Started
+install.packages("devtools")
+devtools::install_github("VangiElia/GEDI4R")
+#loading GEDI4R package
+library(GEDI4R)
+library(magrittr)
+```
+
+```{r}
+## Setting Parameters 
+    country = 'ITA'
+    start_date = "2020-01-01"
+    end_date = "2020-01-31"
+    
+    ## Specific Sites you want to focus on 
+      focus_sites = c("GEDI04_A_2020036151358_O06515_02_T00198_02_002_01_V002.zip",
+                      "GEDI04_A_2021150031254_O13948_03_T06447_02_002_01_V002.zip" )
+      
+    export_csv = TRUE
+    outdir = tempdir
+    ## saved as HDF5 file format 
+      #if we export CSV but didn't provide any path
+      if(as.logical(export_csv) && is.null(outdir)){
+        PEcAn.logger::logger.info("If you want to export CSV file, please ensure input the outdir!")
+        return(0)
+      }
+    outdir = tempdir()
+```
+
+```{r}
+
+```
+
+
+```{r}
+
+ # Using data from GEDI4R package dataset (NASA website)
+  ## Create bounding box (search window) using varname 'country'
+    country_bound <- sf::st_as_sf(raster::getData('GADM', country = country, level = 1))
+         ## raster::getData; get geographic data for anywhere in the world; GADM (database) needs "country ="and level of subdivision 
+    #extract extent
+    e <- raster::extent(country_bound)
+    ## extent creates a matrix 
+    ul_lat <- e@ymax
+    lr_lat <- e@ymin
+    ul_lon <- e@xmin
+    lr_lon <- e@xmax
+    
+    #Get the list of all files available for the study area in the period selected,
+    #using just_path = T
+    file_path <- l4_download(
+      ul_lat,
+      lr_lat,
+      ul_lon,
+      lr_lon,
+      outdir = outdir,
+      from = start_date,
+      to = end_date,
+      just_path = T
+    )
+    
+    #download the first 4 files
+    file_download <- l4_download(
+      ul_lat,
+      lr_lat,
+      ul_lon,
+      lr_lon,
+      ncore = parallel::detectCores()-1,
+      outdir = outdir,
+      from = start_date,
+      to = end_date,
+      just_path = F,
+      subset=1:4
+    )
+    
+      l4_zip <- system.file("extdata",
+                          focus_sites,
+                          package="GEDI4R")
+    
+    #Unzipping GEDI level4A data
+    l4 <- lapply(l4_zip,unzip,exdir = outdir)
+    
+     ## Add columns not previously shown in gediL4 path data set
+    col <-
+      c(
+        "lat_lowestmode",
+        "lon_lowestmode",
+        "delta_time",
+        "agbd_pi_lower",
+        "agbd_pi_upper",
+        "agbd"
+      )
+    
+     gediL4_output <- l4_getmulti(l4,add_col = col, source= T)
+```
+```{r}
+# If using predownloaded dataset (saved as HDF5 file format) - (!) Could be automated to ask or check computer for file paths 
+    site_info = list.files(path =".", pattern = "h5", all.files = T, full.names = T)
```

Comment:
1. earlier you defined site_info as a function input argument, so you don't want to redefine it here
2. in all of the other data constraint functions, site_info provides the site ids, lat, lon, etc. Your function won't work with the larger PEcAn system if you're using the same variable to mean something very different here (vector of hdf5 files)
3. looking for files in the current directory also isn't a general solution. This seems like a folder that should be passed into the function as an argument (e.g. is this outdir? or a subdir off of outdir?)